AY2021S1-CS2103-T16-3/tp
# Extract methods, create enum for keyboard shortcuts, add tests

### What this does
This PR adds credit to sources where certain code is adapted from. Additionally, it adds testing support for the `CreateEditCopy` class. Additionally, it replaces hard-coded keyboard shortcuts in `CommandBox` with `KeyboardShortcutsEnum` as the single source of truth for binding commands together with their keyboard shortcuts, to prevent unnecessary duplication of code dealing with shortcuts in `handleOnKeyPressed`, following the DRY principle.

Edit: Extracted methods `handleShortcuts` and `handleKeyPresses` from `handleOnKeyPressed` for better readability and achieve better modularity, following the SoC principle.


Part of #173 

### How to test

1. Launch the application.
2. Ensure the keyboard shortcuts for `redo`, `undo` and `history` are working (the last is set to <kbd>ctrl-h</kbd>).

### Notes
